### PR TITLE
feat(samples): add example of including httpRequest metadata in log

### PR DESCRIPTION
**Pull request by bcoe**
Demonstrates how to include `httpRequest` metadata in a log-entry:

metadata

_Note: I spruced up our code coverage, because I was eyeballing it while I added a test._

fixes: #240

### CODE REVIEW

File: linkinator.config.json [hunk 1]
```diff
@@ -2,6 +2,7 @@
   "recurse": true,
   "skip": [
     "https://codecov.io/gh/googleapis/",
-    "www.googleapis.com"
+    "www.googleapis.com",
+    "https://github.com/googleapis/nodejs-logging/blob/master/samples/http-request.js"
```
Comment (bcoe) [inline linkinator.config.json]: this is an argument for not generating README.md until the nightly run (although I did find an issue with synthtool generating these READMEs https://github.com/googleapis/synthtool/pull/336).

Comment (JustinBeckwith) [reply linkinator.config.json]: If you have any ideas on how to work around this with linkinator, I'm all ears.  Haven't been able to figure this one out yet :)

File: package.json [hunk 2]
```diff
@@ -30,13 +30,12 @@
     "stackdriver"
   ],
   "scripts": {
-    "cover": "nyc --reporter=lcov mocha build/test/*.js build/test/**/*.js build/test/**/**/*.js && nyc report",
-    "test-no-cover": "mocha build/test/*.js",
-    "test": "npm run cover",
+    "test": "c8 mocha build/test/*.js build/test/**/*.js build/test/**/**/*.js",
```
Comment (JustinBeckwith) [inline package.json]: would `c8 mocha build/test --recurse` be frowned upon?

File: package.json [hunk 3]
```diff
@@ -30,13 +30,12 @@
...
     "docs": "jsdoc -c .jsdoc.js",
     "presystem-test": "npm run compile",
-    "system-test": "mocha build/system-test --timeout 600000",
-    "samples-test": "cd samples/ && npm test && cd ../",
+    "system-test": "c8 mocha build/system-test --timeout 600000",
+    "samples-test": "c8 npm run run-samples-tests",
+    "run-samples-tests": "cd samples/ && npm test && cd ../",
```
Comment (JustinBeckwith) [inline package.json]: This is a new pattern.  If we want to start doing that, should we consider rolling it out broadly and updating the contributing guide?

Comment (bcoe) [reply package.json]: I instead opted for:

```
cd samples/ && c8 npm test && cd ../
```

The same as the original approach, but with the addition of `c8`.

Comment (JustinBeckwith) [reply package.json]: That's great.  I am wondering though - should we consider one hammer PR at this stage to start introducing `c8`?  Maybe cover adding it to all the sample tests at the same time?

File: linkinator.config.json (hunk 1 above)
Comment (bcoe) [reply linkinator.config.json]: I think we should potentially land new samples first, and then let the nightly build add the new links... This only bites us when creating brand new libraries.

**Pull request merged**